ambry-rest/src/main/java/com.github.ambry.rest/AsyncRequestResponseHandler.java

```diff
@@ -317,6 +317,8 @@ public void run() {
           }
         }
       }
+    } catch (Throwable t) {
+      logger.error("AsyncRequestWorker encountered a throwable", t);
```

Comment:
You don't want to rethrow after printing? Also, even if you don't log it here, wouldn't this get printed somewhere higher up?